dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -1712,15 +1711,13 @@ public Entry registerCheckerWorkflow(@ApiParam(hidden = true) @Parameter(hidden
 
             // Determine gitUrl
             gitUrl = tool.getGitUrl();
-
-            // Determine source control, org, and repo
-            Pattern p = Pattern.compile("git@(\\S+):(\\S+)/(\\S+)\\.git");
-            Matcher m = p.matcher(tool.getGitUrl());
-            if (m.find()) {
+            final Optional<Map<String, String>>  stringStringGitUrlMap = SourceCodeRepoFactory
```

Comment:
This is much better that we're not duplicating the parsing any more here.

Reply:
OK